gtfs.js

```diff
@@ -1155,15 +1219,155 @@ class Gtfs {
    *   Map.<string, Map.<string, Object>>
    * } indexedCalendarDates Map of calendarDates properly indexed (see schema.js).
    */
-  setIndexedCalendarDates(indexedCalendarDates) { setIndexedItems(this, 'calendar_dates', indexedCalendarDates); }
+  setIndexedCalendarDates(indexedCalendarDates) { setTable(this, 'calendar_dates', indexedCalendarDates); }
 
 
   /* fare_attributes.txt */
-  // Not used, therefore not implemented
+
+  /**
+   * Adds a fareAttribute in the GTFS.
+   *
+   * @param {Object} fareAttribute - FareAttribute to add in the GTFS.
+   */
+  addFareAttribute(fareAttribute) { addItems(this, 'fare_attributes', [fareAttribute]); }
+
+  /**
+   * Adds a list of fareAttributes in the GTFS.
+   *
+   * @param {Array.<Object>} fareAttributes - Array of fareAttributes to add in the GTFS.
+   */
+  addFareAttributes(fareAttributes) { addItems(this, 'fare_attributes', fareAttributes); }
+
+  /**
+   * Apply a function to each fareAttribute in the GTFS.
+   *
+   * @param {function} iterator - Function which will be applied on every fareAttribute.
+   */
+  forEachFareAttribute(iterator) { forEachItem(this, 'fare_attributes', iterator); }
+
+  /**
+   * Get the indexed fareAttributes of the GTFS. The indexation is defined in the schema (see schema.js).
+   *
+   * @return {Map.<string, Object>} Indexed fareAttributes.
+   */
+  getIndexedFareAttributes() { return getIndexedTable(this, 'fare_attributes'); }
+
+  /**
+   * Get the number of fareAttributes defined
+   *
+   * @returns {number}
+   */
+  getNumberOfFareAttributes() { return getNumberOfItemsInTable(this, 'fare_attributes'); }
+
+  /**
+   * Get the fareAttribute using its fare_id.
+   *
+   * WARNING: Will return the fareAttribute which is indexed with the fare_id passed as argument. If the internal value
+   * of the fareAttribute's fare_id has been changed but not it's indexing, the result will be wrong.
```

Comment:
FYI got bitten by this in an other context on trips (trip_id changed but index not changed). Maybe we should consider having some regular sanity check, at least at export time.